### PR TITLE
Made health analyzer information readout display columns

### PR DESCRIPTION
**Pull request by MarkSuckerberg**
## About The Pull Request
Ike suggested it and it looked easy to add since it was ported directly from /tg/ and looked good.

All credit to /tg/ and coder kingofkosmos for this needed QoL change
## Why It's Good For The Game
Is a lot easier to tell what type of damage is present, and a lot less confusing

LOOK AT THIS SHIT AAAAA

## Changelog
:cl:
tweak: changed the health analyzer damage readout to display columns
/:cl:

### CODE REVIEW

**Comment by ike709** [comment]
At least credit TG...

File: code/game/objects/items/devices/scanners.dm [hunk 1]
```diff
@@ -308,10 +319,10 @@ GENE SCANNER
 
 	// Time of death
 	if(M.tod && (M.stat == DEAD || ((HAS_TRAIT(M, TRAIT_FAKEDEATH)) && !advanced)))
-		to_chat(user, "<span class='info'>Time of Death:</span> [M.tod]")
+		to_chat(user, "<span class='info'>Time of Death: [M.tod]</span>")
 		var/tdelta = round(world.time - M.timeofdeath)
-		if(tdelta < (DEFIB_TIME_LIMIT * 10))
-			to_chat(user, "<span class='danger'>Subject died [DisplayTimeText(tdelta)] ago, defibrillation may be possible!</span>")
+		if(tdelta < (DEFIB_TIME_LIMIT * 10))to_chat(user, "<span class='alert'><b>Subject died [DisplayTimeText(tdelta)] ago, defibrillation may be possible!</b></span>")
```
Comment (ike709) [inline code/game/objects/items/devices/scanners.dm]: You goofed this line.

File: code/game/objects/items/devices/scanners.dm [hunk 2]
```diff
@@ -259,21 +259,32 @@ GENE SCANNER
...
+				<font color='orange'>Burn</font>&#9;\
+				<font color='green'>Toxin</font>&#9;\
+				<font color='purple'>Suffocation</b></font><br>\
+
+				&#9;&nbsp;&nbsp;&nbsp;&nbsp;<span class='info'>Overall:&#9;\
+				<font color='red'>[brute_loss]</font>&#9;\
+				<font color='orange'>[fire_loss]</font>&#9;\
+				<font color='green'>[tox_loss]</font>&#9;\
+				<font color='purple'>[oxy_loss]</font></span><br>"
+			for(var/obj/item/bodypart/org in damaged) //head, left arm, right arm, etc.
```
Comment (ike709) [inline code/game/objects/items/devices/scanners.dm]: Make this a typeless for loop.

**Review by ike709 — CHANGES_REQUESTED** [review]
Do the things.

**Comment by MarkSuckerberg** [comment]
sorry was really tired and rushed it, fixing the things now

**Comment by MarkSuckerberg** [comment]
also had no idea this was a new thing, I thought it was an old forgotten thing that had gotten reverted on /tg/ as a lot of stuff is. Double my bad

File: code/game/objects/items/devices/scanners.dm (hunk 2 above)
Comment (MarkSuckerberg) [reply code/game/objects/items/devices/scanners.dm]: you sure? this is how it is in both the original coder's PR and what is was before. It works as is, if that's the problem

Comment (ike709) [reply code/game/objects/items/devices/scanners.dm]: Typeless loops are faster, and this can be made typeless.

Comment (MarkSuckerberg) [reply code/game/objects/items/devices/scanners.dm]: I'll see what I can do, then

**Comment by MarkSuckerberg** [comment]
also @kingofkosmos hope you're okay with the port, I didn't realize you just made it, I can add credit or something if you want it

File: code/game/objects/items/devices/scanners.dm (hunk 2 above)
Comment (kingofkosmos) [reply code/game/objects/items/devices/scanners.dm]: I think I just made that exact change to my PR, if that helps.

**Comment by qwertyquerty** [comment]
The correct way to give credit would have been to push his commits

**Comment by kingofkosmos** [comment]
Of course! Glad you like it, no need for credit.

**Comment by MarkSuckerberg** [comment]
Alright, thanks man! Didn't mean to go porting something without giving any credit whatsoever...

File: code/game/objects/items/devices/scanners.dm (hunk 2 above)
Comment (MarkSuckerberg) [reply code/game/objects/items/devices/scanners.dm]: Yep, made my job a lot easier by doing that

**Comment by qwertyquerty** [comment]
Fix the stuff

**Comment by MarkSuckerberg** [comment]
Update: Haven't gotten around to fixing this, it all works and I added the typeless loop, but the columns are misaligned. Will fix it whenever I get time.

**Comment by MarkSuckerberg** [comment]
It works now, tested it

**Comment by kingofkosmos** [comment]
FYI I didn't take account that font size can be changed. Here's my quick fix to making the table scale properly: https://github.com/tgstation/tgstation/pull/47095

**Comment by MarkSuckerberg** [comment]
Thanks man for saying something about it, would've missed it otherwise... Good to know about the fix, too!

**Pull request merged**